custom/icds/location_reassignment/tasks.py

```diff
@@ -149,3 +156,33 @@ def process_households_reassignment(domain, reassignments, uploaded_filename, us
             from_email=settings.DEFAULT_FROM_EMAIL
         )
         email.send()
+
+
+@task(queue=settings.CELERY_LOCATION_REASSIGNMENT_QUEUE)
+def process_ucr_changes(domain, case_ids):
+    cases = CaseAccessorSQL.get_cases(case_ids)
+    docs = [case.to_json() for case in cases]
+    data_source_providers = [DynamicDataSourceProvider(), StaticDataSourceProvider()]
+
+    all_configs = [
+        source
+        for provider in data_source_providers
+        for source in provider.by_domain(domain)
+    ]
+
+    adapters = [
+        get_indicator_adapter(config, raise_errors=True, load_source='location_reassignment')
+        for config in all_configs
+    ]
+
+    async_configs_by_doc_id = {}
+    for doc in docs:
```

Comment:
It might be better to add these to the AsyncIndicator table and then queue them with the agg_queue as made possible in https://github.com/dimagi/commcare-hq/pull/27322. That way the processing happens asynchronously, but we still have a way to track its successful completion, and it handles errors relatively gracefully, by leaving them in the queue to be retried

Reply:
@calellowitz are you proposing going back to https://github.com/dimagi/commcare-hq/pull/27186/commits/4b86db439e4ca193e3713c58a91880846f4be936#diff-1acd7eb475e5e2a4ba2528ebaa0f4179R131-R142 ?
and then calling 
`build_indicators_with_agg_queue(list(async_configs_by_doc_id.keys()))`

2 questions.
1. if things actually fail and reprocess later, won't that be bad to us? since we need to finish processing before the aggregation begins on first
2. keeping things in a separate queue here which just deals with what we are doing for location reassignment helps ensure everything is tracked in a single queue which we kind of lose by putting in agg_queue, right?

Reply:
Yeah, that looks right. As far as the questions
1) That would be bad, but the new change doesn't make that more likely. The issue here is that if an error occurs, the celery task will throw an exception but nothing else will happen, and we will have no record of whether the updates succeeded. Storing a list of all the IDs we eventually plan to update somewhere durable, like a sql table, rather than encoded in celery tasks, means if a failure does happen, we will know.
2) I don't think we actually lose info this way. The data we had before, basically just the state of the reassignment queue, is still there its just the agg queue not the reassignment one. The agg queue is unused right now so this won't be shared with anything. Although, if you prefer you can make a new wrapper that puts these in the reassignment queue instead. We do gain additional tracking in the form of the async indicators table this way.